frontend/components/ModelForm.tsx

```diff
@@ -145,92 +146,30 @@ export const ModelTable = ({
-          {objects.length === 0 && (
-            <tr>
-              <td colSpan={tableFields.length + 1}>
-                There are no {noun.toLowerCase()}s to display.
-              </td>
-            </tr>
-          )}
-          {page.map(
-            ({ original: object }, i: number): ReactElement => (
-              <tr key={i}>
-                {tableFields.map((a, i) => (
-                  <td key={i}>
-                    {a.converter
-                      ? a.converter(object[a.name], object)
-                      : object[a.name]}
-                  </td>
-                ))}
-                <td>
-                  <div className="buttons">
-                    {allowEditing && (
-                      <button
-                        onClick={() => onEdit(object)}
-                        className="button is-primary is-small"
-                      >
-                        <Icon name="edit" alt="edit" /> Edit
-                      </button>
-                    )}
-                    {allowDeletion && (
-                      <button
-                        onClick={() => {
-                          if (confirmDeletion) {
-                            if (
-                              confirm(
-                                `Are you sure you want to ${deleteVerb.toLowerCase()} this ${noun.toLowerCase()}?`,
-                              )
-                            ) {
-                              onDelete(object)
-                            }
-                          } else {
-                            onDelete(object)
-                          }
-                        }}
-                        className="button is-danger is-small"
-                      >
-                        <Icon name="trash" alt="delete" /> {deleteVerb}
-                      </button>
-                    )}
-                    {actions && actions(object)}
-                  </div>
-                </td>
-              </tr>
-            ),
-          )}
-        </tbody>
-      </table>
-      {pageOptions.length > 1 && (
-        <div className="is-clearfix">
-          <select
-            value={pageIndex}
-            onChange={(e) => gotoPage(Number(e.target.value))}
-            className="input is-small"
-            style={{ maxWidth: 150 }}
-          >
-            {pageOptions.map(
-              (idx: number): ReactElement => (
-                <option value={idx}>Page {idx + 1}</option>
-              ),
-            )}
-          </select>
-          <span className="is-pulled-right">
-            {objects.length} total entries, {pageSize} entries per page
-          </span>
-        </div>
-      )}
+      <Table
+        data={objects}
+        columns={tableFields}
+        searchableColumns={['name']}
+        filterOptions = {filterOptions}
+        actions = {[{name: "edit", classes:"button is-primary is-small", Icon:"edit", clickFunction:(target) => onEdit(target) } ,
```

Comment:
It might actually be better to pass in a function that returns react elements than this JSON object, because you get more functionality. It isn't too much more additional work, and you get that pure flexibility.

You would want to keep your casing consistent if you used this approach (`Icon` -> `icon`).

Reply:
In fact, it would be better to make this a column and then add the ability to custom render columns. That seems like the cleanest solution.

This way you could have multiple action columns if you wanted, and you would also have the ability to do custom formatting like we do now:
![image](https://user-images.githubusercontent.com/4441174/106333749-4ee15880-6257-11eb-84fd-47a1f0d7328b.png)

We're aiming for a table solution that is flexible enough to replace every table on Penn Clubs! This will make it easier to make/maintain tables going forward.